zetaclient/chains/bitcoin/observer/inbound.go

```diff
@@ -12,63 +12,62 @@ import (
 	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
 
-	"github.com/zeta-chain/node/pkg/coin"
 	"github.com/zeta-chain/node/x/crosschain/types"
 	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/common"
+	"github.com/zeta-chain/node/zetaclient/config"
 	"github.com/zeta-chain/node/zetaclient/logs"
 	"github.com/zeta-chain/node/zetaclient/zetacore"
 )
 
 // ObserveInbound observes the Bitcoin chain for inbounds and post votes to zetacore
-// TODO(revamp): simplify this function into smaller functions
 func (ob *Observer) ObserveInbound(ctx context.Context) error {
-	// get and update latest block height
-	currentBlock, err := ob.rpc.GetBlockCount(ctx)
-	if err != nil {
-		return fmt.Errorf("observeInboundBTC: error getting block number: %s", err)
-	}
-	if currentBlock < 0 {
-		return fmt.Errorf("observeInboundBTC: block number is negative: %d", currentBlock)
+	logger := ob.Logger().Inbound.With().Str(logs.FieldMethod, "ObserveInbound").Logger()
```

Comment:
let's use camel_case for consistency with other scheduler fields
```suggestion
	logger := ob.Logger().Inbound.With().Str(logs.FieldMethod, "observe_inbound").Logger()
```

Reply:
why should we use a `observe_inbound` not matching the true method name?

Reply:
https://github.com/zeta-chain/node/pull/3496/commits/03f6bcc4e8f62b6facf287077753cede38ef6900